### PR TITLE
Replacing deprecated finalize code

### PR DESCRIPTION
**Pull request by Quasar985**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [x] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Replacing `finalize` methods, which have been deprecated for many versions of java, with java's cleaner Api. Cleaners will call the same function as finalize to clean up objects in memory manager.

### Alternate Designs

Phantom references are another solution to replacing finalize. Cleaner was chosen as it requires far less refacotring to work.

### Why Should This Be In Core?

N/A

### Benefits

Deprecated code is removed.

### Possible Drawbacks

If done incorrectly, it may cause memory related issues.

### Verification Process

Run the newly added tests that verify memory manger has removed unused objects.

### Applicable Issues

https://github.com/constellation-app/constellation/issues/1521

### CODE REVIEW

File: CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/locking/DualGraphNGTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.graph.locking;
+
+import au.gov.asd.tac.constellation.utilities.memory.MemoryManager;
+import static org.mockito.ArgumentMatchers.any;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import static org.mockito.Mockito.times;
+import static org.testng.Assert.*;
```
Comment (antares1470) [inline CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/locking/DualGraphNGTest.java:23]: can you replace with specific imports?

File: CoreGraphNode/test/unit/src/au/gov/asd/tac/constellation/graph/node/gui/SimpleGraphTopComponentNGTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ */
+package au.gov.asd.tac.constellation.graph.node.gui;
+
+import au.gov.asd.tac.constellation.graph.node.GraphNode;
+import au.gov.asd.tac.constellation.utilities.memory.MemoryManager;
+import static org.mockito.ArgumentMatchers.any;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import static org.mockito.Mockito.times;
+import static org.testng.Assert.*;
```
Comment (antares1470) [inline CoreGraphNode/test/unit/src/au/gov/asd/tac/constellation/graph/node/gui/SimpleGraphTopComponentNGTest.java:24]: same here

File: CoreGraphNode/toc.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<div class="container">
```
Comment (antares1470) [inline CoreGraphNode/toc.md:1]: I think this file can be removed

**Review by antares1470 — CHANGES_REQUESTED** [review]

File: CoreGraphNode/toc.md (hunk 3 above)
Comment (Quasar985) [reply CoreGraphNode/toc.md]: Removed, not even sure how it made its way in

**Review by Delphinus8821 — APPROVED** [review]

**Review by antares1470 — APPROVED** [review]

**Pull request merged**